tests/testthat/test-render.R

```diff
@@ -276,3 +276,172 @@ test_that("render_website() works - clean render for each language", {
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_book(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org",
+    placement = "navbar"
+  )
+
+  withr::with_dir(parent_dir, render_book(project_dir))
+
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_book", "index.html"))
+  language_links <- xml2::xml_find_first(index, '//div[@id="languages-links-parent"]')
+  navbar_li <- xml2::xml_parent(language_links)
+  navbar_li_class <- xml2::xml_attr(navbar_li, "class")
+
+  expect_equal(navbar_li_class, "nav-item")
+})
+
+test_that("website with sidebar placement has languagelinks in the sidebar", {
+  withr::local_envvar("BABELQUARTO_TESTS_URL" = "true")
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_website(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org",
+    placement = "sidebar"
+  )
+
+  withr::with_dir(parent_dir, render_website(project_dir))
+
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_site", "index.html"))
+  language_links <- xml2::xml_find_first(index, '//div[@id="languages-links-parent"]')
+  sidebar <- xml2::xml_parent(language_links)
+  sidebar_id <- xml2::xml_attr(sidebar, "id")
+
+  expect_equal(sidebar_id, "quarto-sidebar")
+})
+
+test_that("render_book() works - all language links are present", {
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_book(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org"
+  )
+
+  withr::with_dir(parent_dir, render_book(project_dir))
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_book", "index.html"))
+  index_link_fr <- xml2::xml_length(xml2::xml_find_all(index, '//a[@id="language-link-fr"]'), only_elements = FALSE)
+  index_link_es <- xml2::xml_length(xml2::xml_find_all(index, '//a[@id="language-link-es"]'), only_elements = FALSE)
+
+  index_fr <- xml2::read_html(file.path(parent_dir, project_dir, "_book", "fr", "index.fr.html"))
+  index_fr_link_en <- xml2::xml_length(xml2::xml_find_all(index_fr, '//a[@id="language-link-en"]'), only_elements = FALSE)
+  index_fr_link_es <- xml2::xml_length(xml2::xml_find_all(index_fr, '//a[@id="language-link-es"]'), only_elements = FALSE)
+
+  index_es <- xml2::read_html(file.path(parent_dir, project_dir, "_book", "es", "index.es.html"))
+  index_es_link_en <- xml2::xml_length(xml2::xml_find_all(index_es, '//a[@id="language-link-en"]'), only_elements = FALSE)
+  index_es_link_fr <- xml2::xml_length(xml2::xml_find_all(index_es, '//a[@id="language-link-fr"]'), only_elements = FALSE)
+
+  expect_equal(index_link_fr, 1)
```

Comment:
I'd prefer to see the expect_equal closer to the definition of the thing. https://masalmon.eu/2024/05/23/refactoring-tests/#put-the-object-as-close-as-possible-to-the-related-expectations

(you know it's personal taste when you link to your own blog :see_no_evil: )